### PR TITLE
chore: support WG over GRPC in Omni

### PR DESCRIPTION
**Pull request by DmitriyMV**
This PR adds the support for WG over GRPC. New field in `VirtualAddrport` in `SiderolinkSpec` should allow for both setting the virtual addr and loading it after the Omni restart.

### CODE REVIEW

File: internal/pkg/siderolink/manager.go [hunk 1]
```diff
@@ -668,41 +703,73 @@ func (manager *Manager) Provision(ctx context.Context, req *pb.ProvisionRequest)
...
-		ServerAddress:     manager.wgConfig().ServerAddress,
 		NodeAddressPrefix: spec.NodeSubnet,
+		ServerAddress:     manager.wgConfig().ServerAddress,
+		GrpcPeerAddrPort:  spec.VirtualAddrport,
 	}, nil
 }
 
+func tunnelStatusChanged(req *pb.ProvisionRequest, link *siderolink.Link) bool {
+	wgOverGRPC := pointer.SafeDeref(req.WireguardOverGrpc)
+	virtulAddrPort := link.TypedSpec().Value.VirtualAddrport
```
Comment (Unix4ever) [inline internal/pkg/siderolink/manager.go:722]:
```suggestion
	virtualAddrPort := link.TypedSpec().Value.VirtualAddrport
```

**Review by Unix4ever — APPROVED** [review]

File: internal/pkg/siderolink/manager.go (hunk 1 above)
Comment (DmitriyMV) [reply internal/pkg/siderolink/manager.go:722]: 👍

File: internal/pkg/siderolink/manager.go [hunk 2]
```diff
@@ -668,41 +703,73 @@ func (manager *Manager) Provision(ctx context.Context, req *pb.ProvisionRequest)
...
+	switch {
+	case wgOverGRPC && virtulAddrPort == "":
+		return true
+	case !wgOverGRPC && virtulAddrPort != "":
+		return true
+	default:
+		return false
+	}
+}
+
```
[the remark this answers is not in the document]
Comment (DmitriyMV) [reply internal/pkg/siderolink/manager.go:733]: 👍

**Comment by andrewrynhard** [comment]
A very small nit and will come off as annoying I am sure but I would argue this is a `feat`. The idea behind keeping commits accurate is that users and support can look at a CHANGELOG and quickly identify what changed. Features are the most common thing people want to know. A chore is CI related and something people skip.

**Comment by DmitriyMV** [comment]
> A very small nit and will come off as annoying I am sure but I would argue this is a feat. The idea behind keeping commits accurate is that users and support can look at a CHANGELOG and quickly identify what changed. Features are the most common thing people want to know. A chore is CI related and something people skip.

Not at all. But I noticed that too late when I already pressed "/m". :) But I renamed the PR.

**Pull request merged**